### PR TITLE
feat(yaml-config): adding yaml custom configuration

### PR DESCRIPTION
**Pull request by iam-dev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x] Docs
[ ] Other... Please describe:
```

## What is the current behavior?

Adding doc how to add yaml file custom configuration

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: content/techniques/configuration.md [hunk 1]
```diff
@@ -116,6 +116,116 @@ export class AppModule {}
...
+    url: 'localhost'
+    port: 5432
+    database: 'yaml-db'
+    
+  sqlite: 
+    database: 'sqlite.db'
+
+```
+
+To be able to read yaml file and put the configuration in json object, we need to add "js-yaml" to our project.
```
Comment (jmcdo29) [inline content/techniques/configuration.md:140]:
```suggestion
To be able to read yaml file and put the configuration in json object, we need to add `js-yaml` to our project.
```

File: content/techniques/configuration.md [hunk 2]
```diff
@@ -116,6 +116,116 @@ export class AppModule {}
...
+  return JSON.stringify(yamlResult);
+}
+
+```
+
+We are controlling the returned configuration object, so we'll add loadYmlFile() to registerAs. For example:
+
+```typescript
+@@filename(yaml.config.ts)
+import { registerAs } from '../../lib/utils';
```
Comment (jmcdo29) [inline content/techniques/configuration.md:165]: I assume this is meant to be fro m`@nestjs/config`?

File: content/techniques/configuration.md [hunk 3]
```diff
@@ -116,6 +116,116 @@ export class AppModule {}
...
+import { INestApplication } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import { AppModule } from '../src/app.module';
+
+describe('Files', () => {
+  let app: INestApplication;
+
+  beforeEach(async () => {
+    const module = await Test.createTestingModule({
+      imports: [AppModule.withYamlConfigurations()],
```
Comment (jmcdo29) [inline content/techniques/configuration.md:203]: Where does `withYamlConfigurations()` come from?

File: content/techniques/configuration.md [hunk 4]
```diff
@@ -116,6 +116,116 @@ export class AppModule {}
...
+    }).compile();
+
+    app = module.createNestApplication();
+    await app.init();
+  });
+
+  it(`should return loaded configuration (injected through constructor)`, () => {
+    const config = app.get(AppModule).getYamlConfig();
```
Comment (jmcdo29) [inline content/techniques/configuration.md:211]: Similarly, where does `getYamlConfig()` come from?

**Review by jmcdo29 — CHANGES_REQUESTED** [review]

File: content/techniques/configuration.md (hunk 3 above)
Comment (iam-dev) [reply content/techniques/configuration.md:218]: check commit e87b6b1c323d754acf65cfb1c6624ba24295c116

File: content/techniques/configuration.md (hunk 4 above)
Comment (iam-dev) [reply content/techniques/configuration.md:226]: check commit e87b6b1c323d754acf65cfb1c6624ba24295c116

**Review by jmcdo29 — APPROVED** [review]

**Comment by iam-dev** [comment]
I'm not sure why those checks are failling. I'm only changing configuration.md file

**Comment by kamilmysliwiec** [comment]
Thanks!

**Pull request merged**